Source/Core/Core/IOS/IOSC.cpp

```diff
@@ -327,8 +401,20 @@ ReturnCode IOSC::VerifyPublicKeySign(const std::array<u8, 20>& sha1, Handle sign
     mbedtls_mpi_read_binary(&rsa.E, reinterpret_cast<const u8*>(&entry->misc_data), 4);
     rsa.len = entry->data.size();
 
-    const int ret = mbedtls_rsa_pkcs1_verify(&rsa, nullptr, nullptr, MBEDTLS_RSA_PUBLIC,
-                                             MBEDTLS_MD_SHA1, 0, sha1.data(), signature.data());
+    int ret = mbedtls_rsa_pkcs1_verify(&rsa, nullptr, nullptr, MBEDTLS_RSA_PUBLIC, MBEDTLS_MD_SHA1,
+                                       0, sha1.data(), signature.data());
+    if (ret != 0 && m_console_type == ConsoleType::RVT)
+    {
+      // Some dev signatures do not have proper PKCS#1 padding. Just powmod and check it ends with
+      // digest.
+      std::vector<u8> result(signature.size());
+      if (!mbedtls_rsa_public(&rsa, signature.data(), result.data()) &&
```

Comment:
Could we check the result value of `mbedtls_rsa_public` by comparing against 0 explicitly? Makes it more obvious that the function returns an int and not a boolean, and that ret == 0 means success, not false/failure. (I would personally do the same for memcmp but `if (!memcmp(...))` is a more established pattern at least.)